### ISSUE TITLE
stackoverflow in regexp_nfa.c addstate()

### ISSUE THREAD

**Issue by kcwu** [vim/vim#3941]
How to reproduce
```
vim -i NONE -u NONE -Z -e -c 'echo search("\\v((){328}){389}")' -c 'qa!'
```
with latest vim (v8.1.0897)

```
Program received signal SIGSEGV, Segmentation fault.
0x00005555556f537d in addstate (l=0x7fffffffb000, state=0x7fffee089840, subs_arg=0x7fffffffb160, pim=0x0, off_arg=<error reading variable: Cannot access memory at address 0x7fffff7feffc>) at regexp_nfa.c:4295
(gdb) bt 10
#0  0x00005555556f537d in addstate (l=0x7fffffffb000, state=0x7fffee089840, subs_arg=0x7fffffffb160, pim=0x0, off_arg=Python Exception <class 'gdb.MemoryError'> Cannot access memory at address 0x7fffff7feffc:
#1  0x00005555556f57b7 in addstate (l=0x7fffffffb000, state=0x7fffee0897f0, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4497
#2  0x00005555556f5ae3 in addstate (l=0x7fffffffb000, state=0x7fffee089818, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4597
#3  0x00005555556f5e14 in addstate (l=0x7fffffffb000, state=0x7fffee0897c8, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4697
#4  0x00005555556f57b7 in addstate (l=0x7fffffffb000, state=0x7fffee089778, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4497
#5  0x00005555556f5ae3 in addstate (l=0x7fffffffb000, state=0x7fffee0897a0, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4597
#6  0x00005555556f5e14 in addstate (l=0x7fffffffb000, state=0x7fffee089750, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4697
#7  0x00005555556f57b7 in addstate (l=0x7fffffffb000, state=0x7fffee089700, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4497
#8  0x00005555556f5ae3 in addstate (l=0x7fffffffb000, state=0x7fffee089728, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4597
#9  0x00005555556f5e14 in addstate (l=0x7fffffffb000, state=0x7fffee0896d8, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4697
#10 0x00005555556f57b7 in addstate (l=0x7fffffffb000, state=0x7fffee089688, subs_arg=0x7fffffffb160, pim=0x0, off_arg=0) at regexp_nfa.c:4497
```

This is found by fuzzing (libfuzzer)

**Comment by dpelle** [comment]
I suppose that something like `\(\)\{328}` or `\(\)*` could be simplified to `\(\)` since it's a 0-length match.

**Comment by vim-ml** [comment]
Op di 12 feb. 2019 22:15 schreef Bram Moolenaar <vim-dev-github@256bit.org:

> Closed #3941 <https://github.com/vim/vim/issues/3941> via 57ee2b6
> <https://github.com/vim/vim/commit/57ee2b6e0b5b730d12ee9db00a8e2a577df9e374>
> .
>

That commit looks totally unrelated to regular expressions...

Christ van Willegen

>
